GoogleDataTransportCCTSupport/GDTCCTLibrary/GDTCCTUploader.m

```diff
@@ -202,51 +202,39 @@ - (void)uploadPackage:(GDTCORUploadPackage *)package {
         GDTCORLogWarning(GDTCORMCWUploadFailed, @"There was an error uploading events: %@", error);
       }
       NSError *decodingError;
+      GDTCORClock *futureUploadTime;
       if (data) {
         gdt_cct_LogResponse logResponse = GDTCCTDecodeLogResponse(data, &decodingError);
         if (!decodingError && logResponse.has_next_request_wait_millis) {
           GDTCORLogDebug(
               "CCT: The backend responded asking to not upload for %lld millis from now.",
               logResponse.next_request_wait_millis);
-          GDTCORClock *futureUploadTime =
+          futureUploadTime =
               [GDTCORClock clockSnapshotInTheFuture:logResponse.next_request_wait_millis];
-          switch (target) {
-            case kGDTCORTargetCCT:
-              self->_CCTNextUploadTime = futureUploadTime;
-              break;
-
-            case kGDTCORTargetFLL:
-              // Falls through.
-            case kGDTCORTargetCSH:
-              self->_FLLNextUploadTime = futureUploadTime;
-            default:
-              break;
-          }
-        } else {
-          GDTCORLogDebug("%@", @"CCT: The backend response failed to parse, so the next request "
-                               @"won't occur until 15 minutes from now");
-          // 15 minutes from now.
-          GDTCORClock *futureUploadTime = [GDTCORClock clockSnapshotInTheFuture:15 * 60 * 1000];
-          switch (target) {
-            case kGDTCORTargetCCT:
-              self->_CCTNextUploadTime = futureUploadTime;
-              break;
-
-            case kGDTCORTargetFLL:
-              // Falls through.
-            case kGDTCORTargetCSH:
-              self->_FLLNextUploadTime = futureUploadTime;
-              break;
-
-            default:
-              break;
-          }
         }
         pb_release(gdt_cct_LogResponse_fields, &logResponse);
       }
+      if (!futureUploadTime) {
+        GDTCORLogDebug("%@", @"CCT: The backend response failed to parse, so the next request "
+                             @"won't occur until 15 minutes from now");
```

Comment:
drive-by comment: looks like `decodingError` is not checked or logged after calling `GDTCCTDecodeLogResponse`, maybe log it here?

Reply:
Done.